secboot/secboot_tpm.go

```diff
@@ -308,6 +321,27 @@ func unlockEncryptedPartitionWithSealedKey(tpm *sb.TPMConnection, name, device,
 	return nil
 }
 
+// AuthKeyFromKernelKeyring obtains the TPM policy update authorization key
+// for the specified disk and device name from the kernel keyring and unlinks
+// the user keyring from the session keyring.
+func AuthKeyFromKernelKeyring(sourceDevice string) ([]byte, error) {
+	keep := false
+	k, err := sbGetActivationDataFromKernel(keyringPrefix, sourceDevice, keep)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get activation data: %v", err)
+	}
+	if err := unlinkUserKeyring(); err != nil {
+		return nil, fmt.Errorf("cannot unlink user keyring: %v", err)
+	}
+
+	authKey, ok := k.(sb.TPMPolicyAuthKey)
+	if !ok {
+		return nil, fmt.Errorf("internal error: wrong TPM policy auth key format")
```

Comment:
ok will be false in the case of the use of recovery.key  , we need to think how to behave in that case